lib/fs.js

```diff
@@ -211,42 +215,59 @@ Object.defineProperties(Stats.prototype, {
     configurable: true,
     enumerable: true,
     get() {
-      return this._atime !== undefined ?
+      return this._atime !== null ?
           this._atime :
-          (this._atime = new Date(this._atim_msec + 0.5));
+          (this._atime = new Date(this.atimeMs + 0.5));
     },
-    set(value) { return this._atime = value; }
+    set(value) {
+      this.atimeMs = Number(value);
+      return this._atime = value;
+    }
   },
   mtime: {
     configurable: true,
     enumerable: true,
     get() {
-      return this._mtime !== undefined ?
+      return this._mtime !== null ?
           this._mtime :
-          (this._mtime = new Date(this._mtim_msec + 0.5));
+          (this._mtime = new Date(this.mtimeMs + 0.5));
     },
-    set(value) { return this._mtime = value; }
+    set(value) {
+      this.mtimeMs = Number(value);
+      return this._mtime = value;
+    }
   },
   ctime: {
     configurable: true,
     enumerable: true,
     get() {
-      return this._ctime !== undefined ?
+      return this._ctime !== null ?
         this._ctime :
-        (this._ctime = new Date(this._ctim_msec + 0.5));
+        (this._ctime = new Date(this.ctimeMs + 0.5));
     },
-    set(value) { return this._ctime = value; }
+    set(value) {
+      this.ctimeMs = Number(value);
```

Comment:
So you suggest break the connection completely, instead of this one way idiosyncrasy?
... thinking ...
Ok. Makes sense. (also congruent with the **Note:**)